web/libs/editor/src/components/ImageView/ImageView.jsx

```diff
@@ -1281,32 +1281,50 @@ const EntireStage = observer(
 
 const ImageLayer = observer(({ item }) => {
   const imageEntity = item.currentImageEntity;
-  const image = useMemo(() => {
-    const ent = item.currentImageEntity;
+  const konvaImageRef = useRef();
+  const [loadedImage, setLoadedImage] = useState(null);
 
-    if (ent && ent.downloaded) {
+  // Load image with proper CORS and load event
+  useEffect(() => {
+    if (imageEntity?.downloaded && imageEntity.currentSrc) {
       const img = new window.Image();
-      img.src = ent.currentSrc;
-      img.width = Number.parseInt(ent.naturalWidth);
-      img.height = Number.parseInt(ent.naturalHeight);
-      return img;
+      img.crossOrigin = "anonymous";
+      img.src = `${imageEntity.currentSrc}?cb=${Date.now()}`; // avoid tainted cache
+      img.width = Number(imageEntity.naturalWidth);
+      img.height = Number(imageEntity.naturalHeight);
+      img.onload = () => {
+        setLoadedImage(img);
+      };
+    } else {
+      setLoadedImage(null);
     }
-    return null;
-  }, [imageEntity?.downloaded]);
+  }, [imageEntity?.downloaded, imageEntity?.currentSrc]);
 
   const { width, height } = useMemo(() => {
     return {
       width: imageEntity.naturalWidth,
       height: imageEntity.naturalHeight,
     };
-  }, [imageEntity.naturalWidth, imageEntity.naturalHeight, item.stageWidth, item.stageHeight]);
+  }, [imageEntity.naturalWidth, imageEntity.naturalHeight]);
```

Comment:
with the removal item.stageWidth and item.stageHeight dependencies - it would mean this memo would re-fire when that happens - is that expected?
would a window resize potentially trigger these values to change requiring that we update this memoized value?

Reply:
Good catch, thanks